tests/integration/test_phase2_timeout_integration.py

```
@@ -0,0 +1,423 @@
+            start = time.monotonic()
+            embeddings = await provider.embed_batch(["hello"])
+            elapsed = time.monotonic() - start
+        finally:
+            await provider.close()
+
+        # 1 回目 429 → 2 回目 200 の合計 2 リクエスト
+        assert call_count == 2
+        # 復帰したベクトルが返る
+        assert len(embeddings) == 1
+        assert len(embeddings[0]) == 1536
+        # 合計時間境界: per_attempt 2.0s + max_wait 0.1s × (max_attempts-1) = 2.2s 上限。
```

Ruff (0.15.14)

[warning] Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)